### DISCUSSION TITLE
Best fit text with transparent background

### DISCUSSION THREAD

**Discussion by tbmuc** [ImageMagick/ImageMagick#2669] [Help] [answered]
Hello,

I would like to create a transparent png with a best fit text inside and tried the following code with IM 7 and the windows bat shell:
convert -density 300 xc:none -font %FONT%  -fill #3b3935 -size 2336x320 caption:%2  %1-txt.png

The resulting png contains no text. Any idea?

**Comment by snibgo** [comment]
For IM v7, I suggest you use "magick", not "convert".

Your options are in the wrong order.

You create two images. The first is xc:none, just 1x1 pixel. The second is caption:%2. I expect you have two output files. Try this:
```
magick -size 2336x320 -font %FONT% -background None -fill #3b3935 caption:%2 %1-txt.png
```

**Comment by tbmuc** [comment]
Hello snibgo,

thank you very much. Now, I get a text, but that doesn't take use of the full space: http://prntscr.com/usfxpa 
The text should best fit into the box without distortion.

**Comment by snibgo** [comment]
What was your command?

If the pointsize was larger, the base of "y" would be cut off. I expect that the top of some glyphs (that you don't use) would also be cut off.

**Comment by snibgo** [comment, marked as answer]
If you want your text (with just the glyphs you use) to be the maximum size that fits the space, you can make it deliberately too large, then trim, then resize. For example:
```
magick -size 4672x640 -fill #3b3935 caption:"Hello world" -trim +repage -resize 2336x320 x.png
```
The result is 2109x320 pixels. If you need it to be 2336x320, insert "-extent 2336x320" before writing the output.

EDIT: changed "4372" to "4672". I intended the initial size to be twice the required final size. Also added "+repage".
